aten/src/ATen/native/AutogradComposite.cpp

```diff
@@ -22,6 +22,47 @@ std::tuple<at::Tensor, at::Tensor> _unpack_dual(const at::Tensor& tensor, int64_
   return std::tuple<at::Tensor, at::Tensor>(tensor._fw_primal(level), tensor._fw_grad(level));
 }
 
+Tensor _new_zeros_with_same_feature_meta(
+    const at::Tensor& self,
+    const at::Tensor& other,
+    int64_t self_num_batch_dims) {
+  // NB: This function can be called directly from _set_fw_grad or
+  //     if self is batched, from this function's batching rule.
+  auto other_sizes = other.sizes();
+  auto other_strides = other.strides();
+  auto other_storage_offset = other.storage_offset();
+
+  auto self_sizes = self.sizes();
+  auto self_strides = self.strides();
+  auto self_num_feature_dims = self.dim() - self_num_batch_dims;
+
+  // NB: We don't check that the sizes of self is the same as that of other
+  //     because this function is also used in the inplace over view case
+
+  std::vector<int64_t> out_sizes;
+  out_sizes.reserve(self.dim());
+  out_sizes.insert(out_sizes.begin(), other_sizes.begin(), other_sizes.end());
+  out_sizes.insert(out_sizes.begin(), self_sizes.begin(), self_sizes.end() - self_num_feature_dims);
```

Comment:
From checking shouldn't the one above be start at `out_sizes.begin() + self_num_batch_dims` ?

Reply:
Reserve doesn't actually change the size of out_sizes() it only changes capacity, so we still need to insert from 0

Reply:
But now I realize that I reserved the wrong number of elements - it should be `other.dim() + self_num_batch_dims`

Reply:
But also you should first insert the batch dims (using self_sizes) and then at insert at the end the size of other.